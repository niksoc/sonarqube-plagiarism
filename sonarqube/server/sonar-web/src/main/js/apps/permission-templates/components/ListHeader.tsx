/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import HelpTooltip from 'sonar-ui-common/components/controls/HelpTooltip';
import { Alert } from 'sonar-ui-common/components/ui/Alert';
import { translate } from 'sonar-ui-common/helpers/l10n';
import InstanceMessage from '../../../components/common/InstanceMessage';

interface Props {
  organization: T.Organization | undefined;
  permissions: T.Permission[];
}

export default class ListHeader extends React.PureComponent<Props> {
  renderTooltip(permission: T.Permission) {
    return permission.key === 'user' || permission.key === 'codeviewer' ? (
      <div>
        <InstanceMessage message={translate('projects_role', permission.key, 'desc')} />
        <Alert className="spacer-top" variant="warning">
          {translate('projects_role.public_projects_warning')}
        </Alert>
      </div>
    ) : (
      <InstanceMessage message={translate('projects_role', permission.key, 'desc')} />
    );
  }

  render() {
    const cells = this.props.permissions.map(permission => (
      <th
        className="permission-column little-padding-left little-padding-right"
        key={permission.key}>
        <div className="permission-column-inner">
          <span className="text-middle">{translate('projects_role', permission.key)}</span>
          <HelpTooltip className="spacer-left" overlay={this.renderTooltip(permission)} />
        </div>
      </th>
    ));

    return (
      <thead>
        <tr>
          <th className="little-padding-left little-padding-right">&nbsp;</th>
          {cells}
          <th className="thin nowrap text-right little-padding-left little-padding-right">
            &nbsp;
          </th>
        </tr>
      </thead>
    );
  }
}
