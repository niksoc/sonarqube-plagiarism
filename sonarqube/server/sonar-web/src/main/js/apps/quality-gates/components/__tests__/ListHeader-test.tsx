/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import ListHeader from '../ListHeader';

it('should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot();
  const wrapper = shallowRender({ canCreate: true });
  expect(wrapper.find('ModalButton').exists()).toBe(true);
  expect(wrapper.find('ModalButton').dive()).toMatchSnapshot();
});

function shallowRender(props = {}) {
  return shallow(<ListHeader canCreate={false} refreshQualityGates={jest.fn()} {...props} />);
}
