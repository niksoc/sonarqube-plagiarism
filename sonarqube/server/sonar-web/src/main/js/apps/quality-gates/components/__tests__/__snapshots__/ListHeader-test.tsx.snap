// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<header
  className="page-header"
>
  <div
    className="display-flex-center"
  >
    <h1
      className="page-title"
    >
      quality_gates.page
    </h1>
    <DocTooltip
      className="spacer-left"
      doc={Promise {}}
    />
  </div>
</header>
`;

exports[`should render correctly 2`] = `
<Fragment>
  <Button
    data-test="quality-gates__add"
    onClick={[Function]}
  >
    create
  </Button>
</Fragment>
`;
